packages/core/src/node-license/getUserStakedKeyCount.ts

```diff
@@ -0,0 +1,28 @@
+import { ethers } from 'ethers';
+import { config } from '../config.js';
+import { getProvider } from '../utils/getProvider.js';
+import { retry } from "../index.js";
+import { StakingPoolAbi } from '../abis/StakingPoolAbi.js';
+
+/**
+ * Get the staked key count for a user
+ * @param poolAddress - The address of the pool.
+ * @param staker - The address of user that has keys staked.
+ * @returns {number} - Count of staked keys
+ */
+export async function getUserStakedKeyCount(
+    poolAddress: string,
```

Comment:
Will this need to be called multiple times, for each pool a user is staked in?

Or should it accept a user address and return the total staked key count in all pools?

Reply:
This will have to be called for each pool for each owner wallet.

This would be a very good way to optimize RPC costs. I have created a story to think about this should the chance of running the RPC fallback increase over time https://www.pivotaltracker.com/story/show/188104184